File: BookHub.Server/BookHub.Server/Features/Article/Data/Models/Article.cs
namespace BookHub.Server.Features.Article.Data.Models
{
    using System.ComponentModel.DataAnnotations;

    using Server.Data.Models.Base;

    using static Shared.ValidationConstants;

    public class Article : DeletableEntity<int>
    {
        [Required]
        [MaxLength(TitleMaxLength)]
        public string Title { get; init; } = null!;

        [Required]
        [MaxLength(IntroductionMaxLength)]
        public string Introduction { get; init; } = null!;

        [Required]
        [MaxLength(ContentMaxLength)]
        public string Content { get; init; } = null!;

        [MaxLength(UrlMaxLength)]
        public string? ImageUrl { get; set; }

        public int Views { get; set; }
    }
}

File: client/src/api/article/types/article.ts
interface ArticleCore {
  title: string;
  introduction: string;
  imageUrl?: string | null;
}

interface ArticleContent {
  content: string;
}

interface ArticleMetadata {
  id: number;
  createdOn?: string | null;
  views?: number | null;
}

export interface ArticleBase extends ArticleCore, ArticleContent, ArticleMetadata {}

export type Article = ArticleBase;

export interface ArticleSummary extends ArticleCore, Pick<ArticleMetadata, 'id' | 'createdOn'> {}

export interface ArticleInput extends ArticleCore, ArticleContent {}
